tools/ClangPlugin.cpp

```diff
@@ -137,6 +137,20 @@ namespace clad {
       SetRequestOptions(opts);
       DiffCollector collector(DGR, CladEnabledRange, m_DiffRequestGraph, S,
                               opts);
+
+#if CLANG_VERSION_MAJOR > 16
+      for (DiffRequest& request : m_DiffRequestGraph.getNodes()) {
+        if (!request.ImmediateMode ||
+            (!request.Function->isConstexpr() &&
+             !request.Function->isImmediateFunction()))
+          continue;
+
+        m_DiffRequestGraph.setCurrentProcessingNode(request);
+        ProcessDiffRequest(request);
+        m_DiffRequestGraph.markCurrentNodeProcessed();
```

Comment:
```suggestion
      for (DiffRequest& request : m_DiffRequestGraph.getNodes()) {
        if (request.ImmediateMode && 
            (request.Function->isConstexpr() ||
             request.Function->isImmediateFunction())) {
        m_DiffRequestGraph.setCurrentProcessingNode(request);
        ProcessDiffRequest(request);
        m_DiffRequestGraph.markCurrentNodeProcessed();
        }
```

I am not sure why we need to iterate over all of the requests, though...

Reply:
In this method `HandleTopLevelDeclForClad` we are given the new `DeclGroupRef` and then the `DiffCollector` adds the new nodes to the differentiation graph, but we don't really have a way to see what is new. I don't think it would be correct to move the processing part inside the collector as that's not it's purpose. Maybe there should be some way for the collector to give references to just the nodes that it added on it's last traversal? Should I do that or do you have something else in mind?

Reply:
I am wondering if we shouldn't check if there is a `FunctionDecl` in `DGR` and ask it if it were an immediate function?

Reply:
`DGR` is the top level decl, so to find the just the immediate `FunctionDecl`s we'd need to traverse all of it again, which doesn't sound like a great idea, when that's what `DiffCollector` already does, or am I misunderstanding?

Reply:
Ah, yes you are right. Please then just invert the condition as suggested and I think we might be good to go.

Reply:
Okay, I'll do that, but also I think that I'll have some documentation ready by tomorrow, so maybe I can include that in this PR.

Reply:
Agreed. We need a demo but that can be in a separate PR.